Core/src/Detector/CylindricalDetectorHelper.cpp

```diff
@@ -0,0 +1,1115 @@
+/// @param transform The transform of the newly created portal
+/// @param r is the radius of the cylinder
+/// @param zBoundaries the vector of binning boundaries
+/// @param phiBoundaries the vector of binning boundaries
+/// @param index the index of this portal to be set
+/// @param nDir the navigation direction to be set
+///
+/// @return a new portal replacement object
+Acts::Experimental::PortalReplacement createCylinderReplacement(
+    const Acts::Transform3& transform, Acts::ActsScalar r,
+    const std::vector<Acts::ActsScalar>& zBoundaries,
+    const std::vector<Acts::ActsScalar>& phiBoundaries, unsigned int index,
+    Acts::NavigationDirection nDir) {
+  // Autodetector stitch value
+  Acts::BinningValue stitchValue =
+      phiBoundaries.size() == 2u ? Acts::binZ : Acts::binPhi;
+  auto [lengthZ, medZ] = Acts::range_medium(phiBoundaries);
+  auto [sectorPhi, avgPhi] = Acts::range_medium(phiBoundaries);
+
+  // New bounds, with current length and sector values
+  auto bounds = std::make_unique<Acts::CylinderBounds>(r, 0.5 * lengthZ,
+                                                       0.5 * sectorPhi, avgPhi);
+  // A new surface on the negative side over the full range
+  auto surface = Acts::Surface::makeShared<Acts::CylinderSurface>(
+      transform, std::move(bounds));
+
+  // A make a portal and indicate the new link direction
+  const auto& stitchBoundaries =
+      (stitchValue == Acts::binZ) ? zBoundaries : phiBoundaries;
+  return Acts::Experimental::PortalReplacement(
+      Acts::Experimental::Portal::makeShared(surface), index, nDir,
+      stitchBoundaries, stitchValue);
+}
+
+/// @brief Helper function to create a disc portal replacement
+/// @param gcxt the geometry context of thix call
+/// @param volumeCenter a reference center of the volume (combined)
+/// @param refSurface the reference surface (old portal)
+/// @param boundaries the vector of binning boundaries in r
+/// @param binning the binning of the sector (inR, inZ)
+/// @param index the index of this portal to be set
+/// @param nDir the navigation direction to be set
+///
+/// @return a new portal replacement object
+Acts::Experimental::PortalReplacement createSectorReplacement(
+    const Acts::GeometryContext& gctx, const Acts::Vector3& volumeCenter,
+    const Acts::Surface& refSurface,
+    const std::vector<Acts::ActsScalar>& boundaries, Acts::BinningValue binning,
+    unsigned int index, Acts::NavigationDirection nDir) {
+  // Get a reference transform
+  const auto& refTransform = refSurface.transform(gctx);
+  auto refRotation = refTransform.rotation();
+  // Bounds handling
+  const auto& boundValues = refSurface.bounds().values();
+  std::unique_ptr<Acts::PlanarBounds> bounds = nullptr;
+
+  // Create a new transform
+  Acts::Transform3 transform = Acts::Transform3::Identity();
+  if (binning == Acts::binR) {
+    // Center r calculation
+    Acts::ActsScalar centerR =
+        0.5 * (boundaries[0u] + boundaries[boundaries.size() - 1u]);
```

Comment:
could we use `Acts::range_medium(boundaries)` here?

Reply:
here, yes.